video/out/wayland_common.c

```diff
@@ -1651,16 +1655,45 @@ static void info_primaries(void *data, struct wp_image_description_info_v1 *imag
 static void info_primaries_named(void *data, struct wp_image_description_info_v1 *image_description_info,
                                  uint32_t primaries)
 {
+    struct vo_wayland_preferred_description_info *wd = data;
+    if (!wd->is_parametric)
+        return;
+    wd->csp.primaries = map_primaries(primaries);
 }
 
 static void info_tf_power(void *data, struct wp_image_description_info_v1 *image_description_info,
                           uint32_t eexp)
 {
+    struct vo_wayland_preferred_description_info *wd = data;
+    if (!wd->is_parametric)
+        return;
+    // Prefer named values, but try to map pure power curve if we have none.
+    if (wd->csp.transfer != PL_COLOR_TRC_UNKNOWN)
+        return;
+
+    int32_t exp[] = {18000, 20000, 22000, 24000, 26000, 28000};
+    int32_t closest = 0;
+    int32_t min_diff = abs((int32_t)eexp - exp[closest]);
+    for (int i = 1; i < MP_ARRAY_SIZE(exp); ++i) {
+        int diff = abs((int32_t)eexp - exp[i]);
+        if (diff < min_diff) {
+            min_diff = diff;
+            closest = i;
+        }
+    }
+    // Allow 0.1 distance
+    if (min_diff > 1000)
+        return;
```

Comment:
The advantage of using the recommended transfer function is that

1. direct scanout is possible
2. compositing of opaque surfaces can be done with a plain copy

I don't know how much of an advantage choosing an approximate transfer function provides. Maybe it would be better to use the linear transfer function with a floating-point buffer if no exact match exitss.

Reply:
I can remove that. Frankly if there is no named transfer function provided, we would select bt.1886 for sdr, which is already a good choice.

Reply:
It's user choice. There is nothing that can say they have properly calibrated monitor to given gamma. bt.1886 is default for sdr.

Reply:
The user can specify it via `target-trc` or icc.

Reply:
The transfer function chosen by the client has no effect on what is displayed. The choice should be made for efficiency only.

Reply:
Removed map_tf code.

Reply:
Isn't scRGB with linear encoding the recommended transfer?

Reply:
At least on AMD it's not supported, for some reason. On d3d11, on Vulkan it seems to be enumerated. Maybe I will try some day, but it also needs special handling in libplacebo, because it's 80 nits for SDR white (1, 1, 1).